SCons/Node/FS.py

```diff
@@ -1199,8 +1200,9 @@ def __init__(self, path = None):
         DirNodeInfo.fs = self
         FileNodeInfo.fs = self
 
-    def set_SConstruct_dir(self, dir):
+    def set_SConstructs(self, dir, scripts=None):
         self.SConstruct_dir = dir
+        self.SConstructs = scripts
```

Comment:
this change was somewhat specific for the ninja tool, which wants to know the SConstruct files specifically. Technically dir can be computed from Node.SConsctucts[0], but I left the the originally attribute for backwards compatibilty